### ISSUE TITLE
SQL select error

### ISSUE THREAD

**Issue by monetdb-team** [MonetDB/MonetDB#52]
Date: 2004-02-19 12:35:52 +0100
From: Martin Kersten &lt;<mk>&gt;
To:   Niels Nes &lt;<niels>&gt;
Version: -- development

Last updated: 2004-02-25 03:44:01 +0100

## Comment 244
Date: 2004-02-19 12:35:52 +0100
From: Martin Kersten &lt;<mk>&gt;

The script below returns an empty set on the second select
 
 create table tst( a0 int unique, a1 int);
 insert into tst values(1,1);
 insert into tst values(2,2);
 select * from tst;
 select * from tst where a1>=0 and a1 <=3;

## Comment 245
Date: 2004-02-19 12:48:06 +0100
From: Stefan Manegold &lt;<Stefan.Manegold>&gt;

Logged In: YES 
 user_id=572415
 
 found it (though not solved, yet):
 
 with attributes  'b0,b1' instead of 'a0,a1', it works fine:
 ========
 sql> create table tst2( b0 int unique, b1 int);
 sql> insert into tst2 values(1,1);
 sql> insert into tst2 values(2,2);
 sql> select * from tst2;
  b0,   b1  name
  int,  intyp name
  2,      tuplecount
  18,     id
 [ 1,    1   ]
 [ 2,    2       ]
 sql> select * from tst2 where b0>=0 and b1<=3;
  b0,   b1  name
  int,  int  type
  2,      tuplecount
  19,     id
 [ 1,    1       ]
 [ 2,    2       ]
 ========
 
 seem that the column names "a0" & "a1" clash with
 variable/parameter names
 used in the procs that SQL generates for the queries:
 ========
 sql> set explain="mil";
 sql> select * from tst2 where b0>=0 and b1<=3;
  statement  name
  string  type
  23  tuplecount
  20  id
 [ "a0 := sht(\"0\");"   ]
 [ "a1 := sht(\"3\");"   ]
 [ "{"   ]
 [ "var s8 := mvc_bind(myc, \"sys\", \"tst2\", \"b0\", 0);
 tst2"        ]
 [ "var s10 := a0;"      ]
 [ "var s9 := lng(s10);" ]
 [ "var s7 := s8.uselect(s9, lng(nil));" ]
 [ "var s12 := mvc_bind(myc, \"sys\", \"tst2\", \"b1\", 0);
 tst2"       ]
 [ "var s14 := a1;"   ]
 [ "var s13 := lng(s14);"        ]
 [ "var s11 := s12.uselect(lng(nil), s13);"      ]
 [ "var s6 := semijoin(s7, s11);"        ]
 [ "var s15 := oid(\"0@0\");"    ]
 [ "var s5 ark(s6, s15);"    ]
 [ "var s4 := s5.reverse;"       ]
 [ "var s3 4.join(s8);"      ]
 [ "var s16 := s4.join(s12);"    ]
 [ "var s1 := mvc_result_table(myc, 2, s3);"     ]
 [ "mvc_result_column(myc, \"b0\", \"int\", 19, 0, s3);" ]
 [ "mvc_result_column(myc, \"b1\", \"int\", 19, 0, s16);"   
     ]
 [ "mvc_export_result(myc, Output, s1);" ]
 [ "mvc_export_table(myc, Output, s1, 0, int(reply_size));" 
     ]
 [ "}"   ]
 sql> select * from tst where a1>=0 and a1 <=3;
  statement  name
  string  type
  17  tuplecount
  21  id
 [ "a0 := sht(\"0\");"   ]
 [ "a1 := sht(\"3\");"   ]
 [ "{"   ]
 [ "var s7 := mvc_bind(myc, \"sys\", \"tst\", \"a1\", 0);
 tst"  ]
 [ "var s8 := mvc_bind(myc, \"sys\", \"tst\", \"a0\", 0);
 tst"  ]
 
 [ "var s6 := s7.uselect(s8, s7, TRUE, TRUE);"   ]
 
 [ "var s9 := oid(\"0@0\");"     ]
 [ "var s5 := mark(s6, s9);"     ]
 [ "var s4 := s5.reverse;"       ]
 [ "var s3 := s4.join(s8);"      ]
 [ "var s10 := s4.join(s7);"     ]
 [ "var s1 := mvc_result_table(myc, 2, s3);"     ]
 [ "mvc_result_column(myc, \"a0\", \"int\", 19, 0, s3);" ]
 [ "mvc_result_column(myc, \"a1\", \"int\", 19, 0, s10);"   
     ]
 [ "mvc_export_result(myc, Output, s1);" ]
 [ "mvc_export_table(myc, Output, s1, 0, int(reply_size));" 
     ]
 [ "}"   ]
 sql> 
 =========
 
 well, now it's Niels' turn ;-)   
 
 BTW, I expect similar problems in case someone names his
 tables of columns
 s0, s1, s2, ... |-(
 
 in other words, we need something like separate namespaces...
 
 Stefan.
 

## Comment 246
Date: 2004-02-25 03:44:01 +0100
From: Stefan Manegold &lt;<Stefan.Manegold>&gt;

Logged In: YES 
 user_id=572415
 
 Name-space bug has been fixed by Niels.
 
 Related optimizer bug has been fixed by Stefan.
 

## Comment 247
Date: 2005-10-06 10:36:08 +0200
From: Sjoerd Mullender &lt;<sjoerd>&gt;

Logged In: YES 
 user_id=43607
 
 BugDay_2005-10-06: Done.
 test script available as
 src/test/bugs/create_insert_select-bug-sf-900198.sql

## Comment 248
Date: 2010-05-04 09:32:09 +0200
From: Pseudo user for Sourceforge import &lt;<sfimport>&gt;

This bug was previously known as tracker item 900198 at http://sourceforge.net/support/tracker.php?aid=900198
